packages/workspace/src/browser/workspace-service.ts

```diff
@@ -35,25 +39,20 @@ export class WorkspaceService {
         document.title = uri.displayName;
     }
 
-    /**
-     * A promise which will get resolved only and if only when the workspace root URI is set on the backend and it points to an existing directory.
-     */
-    get root(): Promise<FileStat> {
-        return new Promise(async resolve => {
-            const root = await this.server.getRoot();
-            const validRoot = await this.isValidRoot(root);
-            // If the workspace root is either not set or invalid, we never resolve the promise.
-            if (validRoot) {
-                resolve(this.toFileStat(root!));
-            }
-        });
+    onStop(app: FrontendApplication): void {
+        if (this.root) {
+            this.server.setRoot(this.root.uri);
+        }
     }
 
     /**
      * Unlike [root](WorkspaceService.root), this method gets resolved even if the workspace root URI is not yet set on the backend or if it
      * is invalid. In such cases, it will resolve to `undefined`.
      */
     get tryRoot(): Promise<FileStat | undefined> {
+        if (this.root) {
+            return Promise.resolve(this.root);
+        }
```

Comment:
Here, the rest of this property getter never sets the `root` property which is good because it does not have a side-effect, but then the `this.root` always remains `undefined`.

Could you please clean up the documentation, I assume it is not valid anymore.

Reply:
It is set in the constructor, but your question tells me that I should move it here :)

Reply:
Thanks! I believe that would be cleaner. Otherwise, the rest of the code in this getter is just unused.